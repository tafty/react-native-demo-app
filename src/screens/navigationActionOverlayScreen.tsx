import * as R from 'ramda'
import React, { Component } from 'react'
import { ActivityIndicator, Alert, ScrollView, View } from 'react-native'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { Navigation } from 'react-native-navigation'
import * as componentIds from '../componentIds'

import NavigationActionHelper from './navigationActionHelper'

import { NavigationActionOverlayScreenStyles } from '../theme/screens/navigationActionOverlayScreen.style'

const OVERLAY_SHOWS_AFTER_MILLISECONDS = 500
const ERROR_SHOWS_AFTER_MILLISECONDS = 60000

interface IPassedProps {
  componentId: string
  error?: string
  navigationActionId: string
}

interface IMappedProps {}

interface IActionProps {}

interface IProps extends IPassedProps, IMappedProps, IActionProps {}

interface State {
  waitIsSet: boolean
  isLongWait: boolean
}

class NavigationActionOverlayScreen extends Component<IProps, State> {
  timeout?: NodeJS.Timeout
  failureTimeout?: NodeJS.Timeout

  public static propTypes = {
    componentId: PropTypes.string.isRequired,
    error: PropTypes.string,
    navigationActionId: PropTypes.string.isRequired,
  }

  constructor(props: Readonly<IProps>) {
    super(props)

    Navigation.events().bindComponent(this)

    this.state = {
      waitIsSet: false,
      isLongWait: false,
    }
  }

  componentDidAppear() {
    if (!this.state.waitIsSet) {
      this.setState({ waitIsSet: true })
      this.timeout = setTimeout(
        () => this.setState({ isLongWait: true }),
        OVERLAY_SHOWS_AFTER_MILLISECONDS,
      )
      this.failureTimeout = setTimeout(
        () => this._timedOut(),
        ERROR_SHOWS_AFTER_MILLISECONDS,
      )
    }
  }

  componentWillUnmount() {
    if (this.timeout) {
      clearTimeout(this.timeout)
    }

    if (this.failureTimeout) {
      clearTimeout(this.failureTimeout)
    }
  }

  _timedOut() {
    const { error, navigationActionId } = this.props
    // If the navigation overlay has been displayed for more than a minute then something has more than likely gone wrong...
    // ...dismiss the screen...
    try {
      Navigation.dismissOverlay(componentIds.NAVIGATION_ACTION_OVERLAY)
    } catch (e) {}
    // ...cancel the navigation event...

    NavigationActionHelper.complete(navigationActionId)
    /// ...and inform the user
    let message = `Action id: ${navigationActionId}`

    if (!R.isNil(error) && !R.isEmpty(error)) {
      message = error
    }

    Alert.alert('Navigation Time Out', message, [{ text: 'OK' }])
  }

  _renderActivityIndicator = (styles: any) => {
    if (!this.state.isLongWait) {
      return <View />
    }

    return <ActivityIndicator style={styles.activityIndicator} size="large" />
  }

  render() {
    const styles = NavigationActionOverlayScreenStyles.normal
    const scrollContainerStyle = this.state.isLongWait
      ? styles.scrollContainerDark
      : styles.scrollContainer

    return (
      <ScrollView style={scrollContainerStyle}>
        <View style={styles.container}>
          {this._renderActivityIndicator(styles)}
        </View>
      </ScrollView>
    )
  }
}

function mapStateToProps(): IMappedProps {
  return {}
}

const actions = {}

export default connect(mapStateToProps, actions)(NavigationActionOverlayScreen)
