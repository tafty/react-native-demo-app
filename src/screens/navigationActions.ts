export const REPORT_VIEW = 'navigation.actions.REPORT_VIEW'
