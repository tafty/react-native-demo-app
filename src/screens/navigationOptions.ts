import Colors from '../theme/colors'

export class NavigationOptions {
  options: any

  static createNavigationOptions(fixToPortrait: boolean = true) {
    const navigationOptions: NavigationOptions = new NavigationOptions()

    if (fixToPortrait) {
      navigationOptions.options = {
        layout: {
          orientation: ['portrait'],
        },
      }
    } else {
      navigationOptions.options = {
        layout: {
          orientation: ['portrait', 'landscape'],
        },
      }
    }

    navigationOptions.options.animations = {
      setRoot: {
        waitForRender: true,
      },
      showModal: {
        waitForRender: true,
      },
      push: {
        waitForRender: true,
      },
    }

    navigationOptions.options.topBar = {
      drawBehind: true,
      visible: false,
      leftButtonColor: Colors.topBarButtons,
      rightButtonColor: Colors.topBarButtons,
      backButton: {
        color: Colors.topBarButtons,
      },
    }

    return navigationOptions
  }

  addTitle(title: string) {
    if (!this.options.topBar) {
      this.options.topBar = {}
    }

    this.options.topBar.drawBehind = false
    this.options.topBar.visible = true

    if (!this.options.topBar.title) {
      this.options.topBar.title = {}
    }

    this.options.topBar.title.text = title

    return this
  }
}
