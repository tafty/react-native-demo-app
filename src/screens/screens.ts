export const DIAGNOSTICS: string = 'screens.DIAGNOSTICS'
export const HOME: string = 'screens.HOME'
export const LOG_VIEWER: string = 'screens.LOG_VIEWER'
export const LOGIN: string = 'screens.LOGIN'
export const NAVIGATION_ACTION_OVERLAY: string =
  'screen.NAVIGATION_ACTION_OVERLAY'
export const SETTINGS: string = 'screens.SETTINGS'
export const STORYBOOK: string = 'screens.STORYBOOK'
