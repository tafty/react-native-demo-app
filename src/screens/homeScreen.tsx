import * as screens from './screens'

import React, { Component } from 'react'
import { SafeAreaView, ScrollView, Text } from 'react-native'

import type { ApplicationState } from '../reducers'
import Button from '../components/buttons/button'
import { HomeScreenStyles } from '../theme/screens/homeScreen.style'
import { Navigation } from 'react-native-navigation'
import { NavigationOptions } from './navigationOptions'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { createAction } from '../actions/createAction'
import { getConnectionType } from '../selectors/networkConnection/getConnectionType'
import { getIsConnected } from '../selectors/networkConnection/getIsConnected'
import { getIsInternetReachable } from '../selectors/networkConnection/getIsInternetReachable'
import { getRealmPath } from '../selectors/realm/getRealmPath'
import { translate } from '../localisation/translate'

interface IPassedProps {
  componentId: string
}

interface IMappedProps {
  connectionType: string | undefined
  isConnected: boolean
  isInternetReachable: boolean | undefined | null
  realmPath: string
}

interface IActionProps {
  createAction: Function
}

interface IProps extends IPassedProps, IMappedProps, IActionProps {}

class HomeScreen extends Component<IProps> {
  static options() {
    return NavigationOptions.createNavigationOptions().addTitle(
      translate('home'),
    ).options
  }

  public static propTypes = {
    componentId: PropTypes.string.isRequired,
    connectionType: PropTypes.string,
    isConnected: PropTypes.bool.isRequired,
    isInternetReachable: PropTypes.bool,
    createAction: PropTypes.func.isRequired,
    realmPath: PropTypes.string.isRequired,
  }

  _logOutStarted = () => {
    this.props.createAction('LogOutStarted')
  }

  _navigateToSettings = () => {
    Navigation.push(this.props.componentId, {
      component: {
        name: screens.SETTINGS,
      },
    })
  }

  render() {
    const {
      isConnected,
      isInternetReachable,
      connectionType,
      realmPath,
    } = this.props

    const yes = translate('yes')
    const no = translate('no')

    const styles = HomeScreenStyles.normal

    return (
      <SafeAreaView style={styles.container}>
        <ScrollView>
          <Text testID="is_network_connected">
            {translate('is_network_connected')}: {isConnected ? yes : no}
          </Text>
          <Text testID="is_internet_reachable">
            {translate('is_internet_reachable')}
            {': '}
            {isInternetReachable ? yes : no}
          </Text>
          <Text testID="network_connection_type">
            {translate('network_connection_type')}: {connectionType}
          </Text>
          <Text testID="current_realm_path">
            {translate('current_realm_path')}: {realmPath}
          </Text>
          <Button
            onPress={this._navigateToSettings}
            testID="settings-button"
            label={translate('settings')}
            style={styles.button}
          />
          <Button
            onPress={this._logOutStarted}
            testID="log-out-button"
            label={translate('log_out')}
            style={styles.button}
          />
        </ScrollView>
      </SafeAreaView>
    )
  }
}

const actions = {
  createAction,
}

function mapStateToProps(state: ApplicationState): IMappedProps {
  return {
    isConnected: getIsConnected(state),
    isInternetReachable: getIsInternetReachable(state),
    connectionType: getConnectionType(state),
    realmPath: getRealmPath(state),
  }
}

export default connect(mapStateToProps, actions)(HomeScreen)
