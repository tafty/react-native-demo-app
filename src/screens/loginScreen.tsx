import * as R from 'ramda'

import {
  Alert,
  KeyboardAvoidingView,
  Platform,
  SafeAreaView,
  ScrollView,
  Text,
} from 'react-native'
import { ICreateAction, createAction } from '../actions/createAction'
import React, { Component } from 'react'

import type { ApplicationState } from '../reducers'
import Button from '../components/buttons/button'
import LoginInput from '../components/login/loginInput'
import { LoginScreenStyles } from '../theme/screens/loginScreen.style'
import { NavigationOptions } from './navigationOptions'
import PropTypes from 'prop-types'
import ResponseError from '../models/internal/responseError'
import { connect } from 'react-redux'
import { getIsAuthenticating } from '../selectors/security/getIsAuthenticating'
import { getIsConnected } from '../selectors/networkConnection/getIsConnected'
import { getIsInternetReachable } from '../selectors/networkConnection/getIsInternetReachable'
import { getIsOfflineLoginEnabled } from '../selectors/security/getIsOfflineLoginEnabled'
import { getIsOnlineLoginRequired } from '../selectors/security/getIsOnlineLoginRequired'
import { getSecurityError } from '../selectors/security/getSecurityError'
import { getUsername } from '../selectors/security/getUsername'
import moment from 'moment'
import { translate } from '../localisation/translate'

interface IPassedProps {
  componentId: string
}

interface IMappedProps {
  error?: ResponseError | undefined
  isAuthenticating: boolean
  isConnected: boolean
  isInternetReachable: boolean | undefined | null
  isOfflineLoginEnabled: boolean
  isOnlineLoginRequired: boolean
  username?: string | undefined
}

interface IActionProps {
  createAction: ICreateAction
}

interface IProps extends IPassedProps, IMappedProps, IActionProps {}

type State = {
  password: string
  username: string
}

class LoginScreen extends Component<IProps, State> {
  static options() {
    return NavigationOptions.createNavigationOptions().addTitle(
      translate('login'),
    ).options
  }

  public static propTypes = {
    componentId: PropTypes.string.isRequired,
    error: PropTypes.instanceOf(ResponseError),
    isAuthenticating: PropTypes.bool.isRequired,
    isConnected: PropTypes.bool.isRequired,
    isInternetReachable: PropTypes.bool,
    isOnlineLoginRequired: PropTypes.bool.isRequired,
    isOfflineLoginEnabled: PropTypes.bool.isRequired,
    createAction: PropTypes.func.isRequired,
    username: PropTypes.string,
  }

  constructor(props: Readonly<IProps>) {
    super(props)

    const username: string = R.isNil(this.props.username)
      ? ''
      : this.props.username

    this.state = {
      password: '',
      username,
    }
  }

  _isLoginButtonDisabled = () => {
    const {
      isAuthenticating,
      isConnected,
      isInternetReachable,
      isOfflineLoginEnabled,
      isOnlineLoginRequired,
    } = this.props

    const password = this.state.password
    const username = this.state.username

    if (isAuthenticating) {
      return true
    }

    if (R.isNil(password) || R.isEmpty(password)) {
      return true
    }

    if (R.isNil(username) || R.isEmpty(username)) {
      return true
    }

    if (!isConnected || !isInternetReachable) {
      if (isOnlineLoginRequired) {
        return true
      }

      if (!isOfflineLoginEnabled) {
        return true
      }
    }

    return false
  }

  _login = () => {
    this.props.createAction('LoginStarted', {
      username: this.state.username,
      password: this.state.password,
    })
  }

  _onChangePassword = (value: string) => {
    if (typeof value !== 'string') {
      return
    }

    if (R.isNil(value)) {
      value = ''
    }

    this.setState({ password: value })
  }

  _onChangeUsername = (value: string) => {
    if (typeof value !== 'string') {
      return
    }

    if (R.isNil(value)) {
      value = ''
    }

    this.setState({ username: value })
  }

  _onPressForgotPassword = () => {
    Alert.alert('Forgot password', 'Not yet implemented')
  }

  render() {
    const {
      error,
      isAuthenticating,
      isConnected,
      isInternetReachable,
      isOnlineLoginRequired,
    } = this.props

    const hasError = !R.isNil(error)
    // @ts-ignore ramda ensures that error is valid
    const errorMessage = hasError ? error.message : ''
    // @ts-ignore ramda ensures that error is valid
    const errorStatus = hasError ? error.status : ''
    const isLoginButtonDisabled = this._isLoginButtonDisabled()
    const isNetworkErrorVisible =
      isOnlineLoginRequired && (!isConnected || !isInternetReachable)

    const styles = LoginScreenStyles.normal

    return (
      <SafeAreaView style={styles.container}>
        <KeyboardAvoidingView
          behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
          style={styles.keyboardAvoidingView}
        >
          <ScrollView keyboardShouldPersistTaps="always" style={styles.content}>
            <Text style={styles.title}>{translate('login').toUpperCase()}</Text>
            <Text style={styles.instructions}>{translate('instructions')}</Text>
            <LoginInput
              label={translate('username')}
              onChangeText={this._onChangeUsername}
              secureTextEntry={false}
              testID="username"
              value={this.state.username}
            />
            <LoginInput
              label={translate('password')}
              onChangeText={this._onChangePassword}
              secureTextEntry={true}
              style={styles.passwordInput}
              testID="password"
              value={this.state.password}
            />
            <Button
              isDisabled={isLoginButtonDisabled}
              onPress={this._login}
              showActivityIndicator={isAuthenticating}
              style={styles.loginButton}
              testID="login-button"
              label={translate('login')}
            />
            {hasError && (
              <Text style={styles.error} testID="error_message">
                {translate(errorMessage, { status: errorStatus })}
              </Text>
            )}
            {isNetworkErrorVisible && (
              <Text style={styles.network} testID="is_network_connected">
                {translate('no_internet_connection')}
              </Text>
            )}
            <Text
              onPress={this._onPressForgotPassword}
              style={styles.forgotPassword}
              testID="forgot_password"
            >
              {translate('forgot_password')}
            </Text>
            <Text style={styles.copyright}>
              {translate('copyright', {
                year: moment().format('YYYY'),
              })}
            </Text>
          </ScrollView>
        </KeyboardAvoidingView>
      </SafeAreaView>
    )
  }
}

const actions: IActionProps = {
  createAction,
}

function mapStateToProps(state: ApplicationState): IMappedProps {
  return {
    error: getSecurityError(state),
    isAuthenticating: getIsAuthenticating(state),
    isConnected: getIsConnected(state),
    isInternetReachable: getIsInternetReachable(state),
    isOfflineLoginEnabled: getIsOfflineLoginEnabled(state),
    isOnlineLoginRequired: getIsOnlineLoginRequired(state),
    username: getUsername(state),
  }
}

export default connect(mapStateToProps, actions)(LoginScreen)
