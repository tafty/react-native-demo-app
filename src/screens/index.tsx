import React from 'react'
import { Navigation } from 'react-native-navigation'

import HomeScreen from './homeScreen'
import LoginScreen from './loginScreen'
import DiagnosticsScreen from './settings/diagnosticsScreen'
import LogViewerScreen from './settings/logViewerScreen'
import NavigationActionOverlayScreen from './navigationActionOverlayScreen'
import SettingsScreen from './settings/'
import StorybookScreen from './settings/storybook'

import * as screens from './screens'

function registerScreenWithRedux(
  screenName: string,
  ReduxScreen: React.ComponentType<any>,
  store: Object,
  Provider: React.ComponentType<any>,
) {
  Navigation.registerComponent(
    screenName,
    () => props => (
      <Provider store={store}>
        <ReduxScreen {...props} />
      </Provider>
    ),
    () => ReduxScreen,
  )
}

export function registerScreens(
  store: Object,
  provider: React.ComponentType<any>,
) {
  registerScreenWithRedux(
    screens.DIAGNOSTICS,
    DiagnosticsScreen,
    store,
    provider,
  )
  registerScreenWithRedux(screens.HOME, HomeScreen, store, provider)
  registerScreenWithRedux(screens.LOGIN, LoginScreen, store, provider)
  registerScreenWithRedux(screens.LOG_VIEWER, LogViewerScreen, store, provider)
  registerScreenWithRedux(
    screens.NAVIGATION_ACTION_OVERLAY,
    NavigationActionOverlayScreen,
    store,
    provider,
  )
  registerScreenWithRedux(screens.SETTINGS, SettingsScreen, store, provider)
  registerScreenWithRedux(screens.STORYBOOK, StorybookScreen, store, provider)
}
