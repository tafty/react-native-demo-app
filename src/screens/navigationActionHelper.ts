import * as componentIds from '../componentIds'
import * as screens from './screens'

import { Navigation } from 'react-native-navigation'

var navigationActions: Array<string> = []

/**
 *
 * Use this helper to show an activity indicator on an overlay when an action is dispatched
 * from a UI component that may result in a delay to navigation occurring
 *
 * For example, if dispatching an action results in a saga interacting with the database
 * and then updating state followed by the actual navigation occurring
 *
 * The action will need to include a navigationId and componentId in it's parameters
 *
 * From the UI component dispatching the action call:
 *
 *    NavigationActionHelper.start(navigationActions.EDIT, () =>
 *      this.props.editAction(navigationActions.EDIT, this.props.componentId, data)
 *    )
 *
 * In the saga, once all processing is complete and navigation is to occur call:
 *
 *    NavigationActionHelper.navigate(
 *      action.navigationId,
 *      action.componentId,
 *      screens.NEXT_SCREEN
 *    )
 *
 * Alternatively, if navigation needs to be cancelled call:
 *
 *    NavigationActionHelper.complete(navigationActions.EDIT)
 *
 * *** THIS MECHANISM SHOULD NOT BE USED FOR VERY LONG RUNNING PROCESSES SUCH AS NETWORK CALLS ***
 *
 */
export default class NavigationActionHelper {
  static reset = () => {
    navigationActions = []
  }

  static start = (
    navigationActionId: string,
    method: Function,
    error?: string,
  ) => {
    if (navigationActions.indexOf(navigationActionId) > -1) {
      return
    }

    navigationActions.push(navigationActionId)

    Navigation.showOverlay({
      component: {
        id: componentIds.NAVIGATION_ACTION_OVERLAY,
        name: screens.NAVIGATION_ACTION_OVERLAY,
        passProps: {
          error: error,
          navigationActionId,
        },
        options: {
          layout: {
            componentBackgroundColor: 'transparent',
          },
          overlay: {
            interceptTouchOutside: true,
          },
          animations: {
            showModal: {
              enabled: false,
            },
          },
        },
      },
    }).then(() => method())
  }

  static navigate = (
    navigationActionId: string,
    componentId: string,
    screenId: string,
    options?: any,
    passProps?: any,
    id?: string,
  ) => {
    var layout = {
      component: {
        name: screenId,
        options,
        passProps,
        id,
      },
    }

    Navigation.push(componentId, layout).then(() => {
      NavigationActionHelper.complete(navigationActionId)
    })
  }

  static complete = (navigationActionId: string) => {
    if (navigationActions.indexOf(navigationActionId) > -1) {
      try {
        Navigation.dismissOverlay(componentIds.NAVIGATION_ACTION_OVERLAY)
      } catch (e) {}

      navigationActions = navigationActions.filter(event => {
        return event !== navigationActionId
      })
    }
  }
}
