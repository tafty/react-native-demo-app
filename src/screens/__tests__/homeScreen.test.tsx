import 'react-native'
import React from 'react'
import { Navigation } from 'react-native-navigation'
import { render, fireEvent } from 'react-native-testing-library'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import HomeScreen from '../homeScreen'
import * as screens from '../screens'

const mockStore = configureStore()

describe('Home Screen', () => {
  beforeEach(() => {
    Navigation.push = jest.fn()
  })

  it('Should render when network connected', () => {
    const componentId = 'component1'
    const connectionType = 'wifi'
    const environment = 'test'
    const initialState = {
      configuration: {
        localConfig: {
          environment,
        },
      },
      networkConnection: {
        isConnected: true,
        isInternetReachable: true,
        connectionType,
      },
    }

    const store = mockStore(initialState)

    const { getByTestId, toJSON } = render(
      <Provider store={store}>
        <HomeScreen
          componentId={componentId}
          isConnected={false}
          createAction={() => {}}
          realmPath=""
        />
      </Provider>,
    )

    expect(getByTestId('is_network_connected').props.children).toMatchObject([
      'is_network_connected',
      ': ',
      'yes',
    ])
    expect(getByTestId('is_internet_reachable').props.children).toMatchObject([
      'is_internet_reachable',
      ': ',
      'yes',
    ])
    expect(
      getByTestId('network_connection_type').props.children,
    ).toMatchObject(['network_connection_type', ': ', connectionType])
    expect(getByTestId('current_realm_path').props.children).toMatchObject([
      'current_realm_path',
      ': ',
      environment + '.realm',
    ])

    fireEvent.press(getByTestId('settings-button'))
    const push = Navigation.push as jest.Mock
    expect(push.mock.calls).toHaveLength(1)
    expect(push.mock.calls[0][1].component.name).toEqual(screens.SETTINGS)

    expect(toJSON()).toMatchSnapshot()
  })

  it('Should render values when network not connected', () => {
    const componentId = 'component1'
    const connectionType = 'none'
    const environment = 'test2'
    const initialState = {
      configuration: {
        localConfig: {
          environment,
        },
      },
      networkConnection: {
        isConnected: false,
        isInternetReachable: false,
        connectionType,
      },
    }

    const store = mockStore(initialState)

    const { getByTestId, toJSON } = render(
      <Provider store={store}>
        <HomeScreen
          componentId={componentId}
          isConnected={false}
          createAction={() => {}}
          realmPath=""
        />
      </Provider>,
    )

    expect(getByTestId('is_network_connected').props.children).toMatchObject([
      'is_network_connected',
      ': ',
      'no',
    ])
    expect(getByTestId('is_internet_reachable').props.children).toMatchObject([
      'is_internet_reachable',
      ': ',
      'no',
    ])
    expect(
      getByTestId('network_connection_type').props.children,
    ).toMatchObject(['network_connection_type', ': ', connectionType])
    expect(getByTestId('current_realm_path').props.children).toMatchObject([
      'current_realm_path',
      ': ',
      environment + '.realm',
    ])

    fireEvent.press(getByTestId('settings-button'))
    const push = Navigation.push as jest.Mock
    expect(push.mock.calls).toHaveLength(1)
    expect(push.mock.calls[0][1].component.name).toEqual(screens.SETTINGS)

    expect(toJSON()).toMatchSnapshot()
  })
})
