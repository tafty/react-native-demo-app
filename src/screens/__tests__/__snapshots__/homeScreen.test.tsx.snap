// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Should render values when network not connected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "marginBottom": 32,
      "marginHorizontal": 32,
      "marginTop": 24,
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        testID="is_network_connected"
      >
        is_network_connected
        : 
        no
      </Text>
      <Text
        testID="is_internet_reachable"
      >
        is_internet_reachable
        : 
        no
      </Text>
      <Text
        testID="network_connection_type"
      >
        network_connection_type
        : 
        none
      </Text>
      <Text
        testID="current_realm_path"
      >
        current_realm_path
        : 
        test2.realm
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginTop": 16,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 124, 255, 1)",
              "borderColor": "rgba(0, 124, 255, 1)",
              "borderRadius": 4,
              "borderWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 16,
            }
          }
          testID="settings-button"
        >
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "flex": 0,
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
            testID="settings-button-label"
          >
            SETTINGS
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginTop": 16,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 124, 255, 1)",
              "borderColor": "rgba(0, 124, 255, 1)",
              "borderRadius": 4,
              "borderWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 16,
            }
          }
          testID="log-out-button"
        >
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "flex": 0,
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
            testID="log-out-button-label"
          >
            LOG_OUT
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Home Screen Should render when network connected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "marginBottom": 32,
      "marginHorizontal": 32,
      "marginTop": 24,
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        testID="is_network_connected"
      >
        is_network_connected
        : 
        yes
      </Text>
      <Text
        testID="is_internet_reachable"
      >
        is_internet_reachable
        : 
        yes
      </Text>
      <Text
        testID="network_connection_type"
      >
        network_connection_type
        : 
        wifi
      </Text>
      <Text
        testID="current_realm_path"
      >
        current_realm_path
        : 
        test.realm
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginTop": 16,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 124, 255, 1)",
              "borderColor": "rgba(0, 124, 255, 1)",
              "borderRadius": 4,
              "borderWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 16,
            }
          }
          testID="settings-button"
        >
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "flex": 0,
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
            testID="settings-button-label"
          >
            SETTINGS
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginTop": 16,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 124, 255, 1)",
              "borderColor": "rgba(0, 124, 255, 1)",
              "borderRadius": 4,
              "borderWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 16,
            }
          }
          testID="log-out-button"
        >
          <Text
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "flex": 0,
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
            testID="log-out-button-label"
          >
            LOG_OUT
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
